File: day1/first/main.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"slices"
)

func main() {
	llist := make([]int, 1000)
	rlist := make([]int, 1000)
	diff := make([]int, 1000)

	input, err := os.Open("../input.txt")
	scanner := bufio.NewScanner(input)
	if err != nil {
		os.Exit(1)
	}
	defer input.Close()

	i := 0
	for scanner.Scan() {
		line := scanner.Text()
		fmt.Sscanf(line, "%d %d", &llist[i], &rlist[i])
		i++
	}

	slices.Sort(llist)
	slices.Sort(rlist)

	for i := range len(llist) {
		diff[i] = int(math.Abs(float64(llist[i] - rlist[i])))
	}

	sum := 0
	for val := range slices.Values(diff) {
		sum += val
	}
	fmt.Println(sum)
}

File: day1/first/main.py
llist = [0] * 1000
rlist = [0] * 1000

with open("../input.txt", "r") as file:
    i = 0
    while (line := file.readline()) and line:
        llist[i], rlist[i] = map(int, line.split())
        i += 1

llist.sort()
rlist.sort()

diff = [abs(left - right) for left, right in zip(llist, rlist)]
print(sum(diff))